src/cast_core/src/CollectionScanner.cpp

```diff
@@ -227,9 +241,117 @@ void countScan(CollectionScanner::PhaseConfig* config,
     statTracker.success();
 }
 
+void pointInTimeScan(mongocxx::pool::entry& client,
+                     CollectionScanner::PhaseConfig* config,
+                     std::optional<bsoncxx::types::b_timestamp> readClusterTime) {
+    int64_t docCount = 0;
+    int64_t scanSize = 0;
+    bool scanFinished = false;
+    mongocxx::options::client_session sessionOption;
+    sessionOption.causal_consistency(false);
+    mongocxx::client_session session = client->start_session(sessionOption);
+    if (!readClusterTime) {
+        client->database("admin").run_command(session, make_document(kvp("ping", 1)));
+        readClusterTime = session.operation_time();
+    }
+    bsoncxx::document::value readConcern = bson_stream::document{}
+        << "level"
+        << "snapshot"
+        << "atClusterTime" << readClusterTime.value() << bson_stream::finalize;
```

Comment:
here readClusterTime could be an empty optional

Reply:
If readClusterTime is empty, we use session.operation_time() to initialize it at line 255. And that's expected to return a non-empty timestamp.